tests/loggers/test_wandb_logger.py

```diff
@@ -76,6 +85,113 @@ def test_wandb_ml_log_image_errors_out(test_wandb_logger, images, channels_last)
+                                                            'pred': torch.randint(0, 10, (32, 32))
+                                                        }, False),
+                                                        (torch.rand(4, 3, 32, 32), {
+                                                            'pred': torch.randint(0, 10, (4, 32, 32))
+                                                        }, False),
+                                                        (torch.rand(4, 3, 32, 32), {
+                                                            'pred': torch.randint(0, 10, (4, 32, 32)),
+                                                            'pred2': torch.randint(0, 10, (4, 32, 32))
+                                                        }, False)])
+def test_wandb_log_image_with_masks(images, masks, test_wandb_logger, channels_last):
+    pytest.importorskip('wandb', reason='wandb is optional')
+
+    num_masks = len(masks.keys())
+    expected_num_images = 1 if images.ndim < 4 else images.shape[0]
+    expected_num_masks = num_masks * expected_num_images
+
+    test_wandb_logger.log_images(images=images, masks=masks, channels_last=channels_last)
+    test_wandb_logger.post_close()
+    img_dir = str(Path(test_wandb_logger.run_dir) / Path('media/images'))
+    imgs = [
+        filename for filename in os.listdir(img_dir)
+        if not os.path.isdir(img_dir + '/' + filename) and imghdr.what(img_dir + '/' + filename) == 'png'
+    ]
+    actual_num_images = len(imgs)
+    assert actual_num_images == expected_num_images
+
+    mask_dir = str(Path(test_wandb_logger.run_dir) / Path('media/images/mask'))
+    msks = [filename for filename in os.listdir(mask_dir) if imghdr.what(mask_dir + '/' + filename) == 'png']
+    actual_num_masks = len(msks)
+    assert actual_num_masks == expected_num_masks
+
+
+@pytest.mark.parametrize('images,masks', [(torch.randint(0, 256, (32, 32, 3)), {
+    'pred': torch.randint(0, 10, (32, 32))
+})])
+def test_wandb_log_image_with_masks_and_table(monkeypatch, images, masks, test_wandb_logger):
+    wandb = pytest.importorskip('wandb', reason='wandb is optional')
+
+    expected_num_images = 1 if images.ndim < 4 else images.shape[0]
+
+    assert wandb.run is not None
+    wandb_run_dir = Path(wandb.run.dir)
+    test_wandb_logger.log_images(images=images, masks=masks, channels_last=True, use_table=True)
+    test_wandb_logger.post_close()
+
+    wandb_media_dir = wandb_run_dir.parent / Path('files') / Path('media') / Path('table')
+    image_table_files = wandb_media_dir.glob('./*.json')
+
+    image_count = 0
+    for image_table_file in image_table_files:
+        table_columns = json.load(open(image_table_file.absolute()))['data']
+        num_images = sum([1 for column in table_columns if column[0] == 'Image'])
+        image_count += num_images
+
+    assert image_count == expected_num_images
+
+
+def test_wandb_and_image_visualizer(tmp_path, test_wandb_logger):
```

Comment:
Are the image visualizer tests in the test logger files or in the test image visualizer file? It seems like its in both places right now?

Reply:
just misnamed